### ISSUE TITLE
iOS Share sheet callback

### ISSUE THREAD

**Issue by zanemcca** [BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK#144]
We should have a callback from the share sheet for iOS as well as android. I need to be able to run additional code only after a successful share.

**Comment by aaustin** [comment]
Hey @zanemcca - Just to give you an update, we're adding this feature in now and will submit a PR today or tomorrow. Thanks for the request!

**Comment by zanemcca** [comment]
Awesome thanks a lot @aaustin

**Comment by renesansz** [comment]
Hey @zanemcca ,
Just have submitted PR #145 for the iOS share sheet callback.
Let me know if this resolves the issue. 👍

**Comment by rklus16** [comment]
👍

**Comment by zanemcca** [comment]
Hey @renesansz ,
Thanks a lot for the quick response. I had a look and it is almost there except we need a corresponding success callback to go along with the dismiss. There is a [completed flag](https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/blob/5618ae04c8f358159a2206ac8c16c99df9dd2d52/src/ios/BranchSDK.m#L482) that could be used to distinguish between the two.

Could it be possible to call `onShareSheetDismissed` for `completed = False` and then call `onLinkShareResponse` with the `activityType` passed as the result otherwise?

**Comment by zanemcca** [comment]
Or just match the functionality of Android.

I am not really sure why there are so many different callbacks in the first place. Why not just return a promise that is resolved if there is a `sharedChannel/activityType` and reject it otherwise? 
This would also make it match the `generateShortUrl` syntax.

**Comment by renesansz** [comment]
@zanemcca ,
I see, I have taken a look with iOS SDK but I'm not sure if its share sheet callbacks are consistent with Android SDK. Sure, I'll try to make some changes and update the PR 👍 .

**Comment by renesansz** [comment]
@zanemcca , I've made some additional updates in the PR, however, the callbacks aren't complete as the Android (missing: `onChannelSelected` & `onShareSheetLaunched`).
Let me know of your thoughts 👌

**Comment by zanemcca** [comment]
@renesansz ,
Thanks again for the speedy reply. I took a look and everything seems to be functioning properly.

The last thing that would be great to have is if the format of the response was identical to androids.

```
{
  sharedLink: 'https://blahblah.com/1234/asdfjknasdkfjn',
  sharedChannel: 'com.google.Gmail.ShareExtension'
}
```

Having this would make handling the response identical for both platforms and it gives us access to the `sharedLink` and `sharedChannel` which are both nice things to save.

Sorry for being picky. I would do it myself but my Objective-C knowledge is basically non-existent.

**Comment by renesansz** [comment]
@zanemcca ,
Sure, I'll update the response being returned from the callback.
No worries 👌

**Comment by renesansz** [comment]
@zanemcca , Updated PR to return `sharedLink` & `sharedChannel` for the callback response.
Let me know if this will satisfy your needs.

**Comment by zanemcca** [comment]
Thanks a lot @renesansz!
That will definitely be enough to satisfy my needs 👍

**Comment by aaustin** [comment]
Merged and deployed https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/pull/145
